core/src/main/java/com/linecorp/armeria/server/cors/AbstractCorsPolicyBuilder.java

```diff
@@ -0,0 +1,383 @@
+        return self();
+    }
+
+    /**
+     * Specifies the headers to be exposed to calling clients.
+     *
+     * <p>During a simple CORS request, only certain response headers are made available by the
+     * browser, for example using:
+     * <pre>{@code
+     * xhr.getResponseHeader("Content-Type");
+     * }</pre>
+     *
+     * <p>The headers that are available by default are:
+     * <ul>
+     * <li>Cache-Control</li>
+     * <li>Content-Language</li>
+     * <li>Content-Type</li>
+     * <li>Expires</li>
+     * <li>Last-Modified</li>
+     * <li>Pragma</li>
+     * </ul>
+     *
+     * <p>To expose other headers they need to be specified which is what this method enables by
+     * adding the headers to the CORS 'Access-Control-Expose-Headers' response header.
+     *
+     * @param headers the values to be added to the 'Access-Control-Expose-Headers' response header
+     * @return {@link B} to support method chaining.
+     */
+    public B exposeHeaders(final AsciiString... headers) {
+        requireNonNull(headers, "headers");
+        for (int i = 0; i < headers.length; i++) {
+            if (headers[i] == null) {
+                throw new NullPointerException("headers[" + i + ']');
+            }
+        }
+        Collections.addAll(exposedHeaders, headers);
+        return self();
+    }
+
+    /**
+     * Specifies the allowed set of HTTP Request Methods that should be returned in the
+     * CORS 'Access-Control-Request-Method' response header.
+     *
+     * @param methods the {@link HttpMethod}s that should be allowed.
+     * @return {@link B} to support method chaining.
+     */
+    public B allowRequestMethods(final HttpMethod... methods) {
+        requireNonNull(methods, "methods");
+        for (int i = 0; i < methods.length; i++) {
+            if (methods[i] == null) {
+                throw new NullPointerException("methods[" + i + ']');
+            }
+        }
+        Collections.addAll(allowedRequestMethods, methods);
+        return self();
+    }
+
+    /**
+     * Specifies the if headers that should be returned in the CORS 'Access-Control-Allow-Headers'
+     * response header.
+     *
+     * <p>If a client specifies headers on the request, for example by calling:
+     * <pre>{@code
+     * xhr.setRequestHeader('My-Custom-Header', "SomeValue");
+     * }</pre>
+     * the server will receive the above header name in the 'Access-Control-Request-Headers' of the
+     * preflight request. The server will then decide if it allows this header to be sent for the
+     * real request (remember that a preflight is not the real request but a request asking the server
+     * if it allow a request).
+     *
+     * @param headers the headers to be added to the preflight 'Access-Control-Allow-Headers' response header.
+     * @return {@link B} to support method chaining.
+     */
+    public B allowRequestHeaders(final String... headers) {
```

Comment:
nit: How about changing `String` to `CharSequence` and then removing `public B allowRequestHeaders(final AsciiString... headers)`? We can use cached `AsciiString` for both of `String` and `AsciiString`.

Reply:
you're right. Thanks!